src/platform_impl/windows/window.rs

```diff
@@ -389,23 +389,49 @@ impl Window {
     }
 
     #[inline]
-    pub fn fullscreen(&self) -> Option<RootMonitorHandle> {
+    pub fn fullscreen(&self) -> Option<Fullscreen> {
         let window_state = self.window_state.lock();
         window_state.fullscreen.clone()
     }
 
     #[inline]
-    pub fn set_fullscreen(&self, monitor: Option<RootMonitorHandle>) {
+    pub fn set_fullscreen(&self, fullscreen: Option<Fullscreen>) {
         unsafe {
             let window = self.window.clone();
             let window_state = Arc::clone(&self.window_state);
 
-            match &monitor {
-                &Some(RootMonitorHandle { ref inner }) => {
-                    let (x, y): (i32, i32) = inner.position().into();
-                    let (width, height): (u32, u32) = inner.size().into();
+            match fullscreen {
+                None => {
+                    self.thread_executor.execute_in_thread(move || {
+                        let mut window_state_lock = window_state.lock();
+                        window_state_lock.fullscreen = None;
+
+                        if let Some(SavedWindow{client_rect, dpi_factor}) = window_state_lock.saved_window {
+                            window_state_lock.dpi_factor = dpi_factor;
+                            window_state_lock.saved_window = None;
+
+                            WindowState::refresh_window_state(
+                                window_state_lock,
+                                window.0,
+                                Some(client_rect)
+                            );
+                        }
+
+                        mark_fullscreen(window.0, false);
+                        return;
+                    });
+                }
+                Some(fullscreen) => {
+                    let (position, size): ((i32, i32), (u32, u32)) = match fullscreen {
+                        Fullscreen::Exclusive(ref video_mode) => (
+                            video_mode.monitor().position().into(),
```

Comment:
Right now, this retrieves the monitor's position before the video mode transition. What ends up happening is that (at least on my multi-monitor setup), the monitor's position ends up changing after the video mode changes, so the window gets set to the wrong position. We should be retrieving this value after we've changed the video mode.

Reply:
Interesting! This didn't seem to occur on my system. I'll take a look.